File: editor/en/[2225]Find Players With Zero or One Losses.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Map<Integer, Integer> loseCount = new HashMap<>();

        for (int[] match : matches) {
            int winner = match[0];
            int loser = match[1];
            loseCount.put(winner, loseCount.getOrDefault(winner, 0));
            loseCount.put(loser, loseCount.getOrDefault(loser, 0) + 1);
        }

        List<Integer> noLose = new ArrayList<>();
        List<Integer> oneLose = new ArrayList<>();

        for (int player : loseCount.keySet()) {
            if (loseCount.get(player) == 0) {
                noLose.add(player);
            } else if (loseCount.get(player) == 1) {
                oneLose.add(player);
            }
        }

        List<List<Integer>> result = new ArrayList<>();
        Collections.sort(noLose);
        Collections.sort(oneLose);
        result.add(noLose);
        result.add(oneLose);
        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2225]Find Players With Zero or One Losses.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function (matches) {
    const loseCount = new Map();

    for (const [winner, loser] of matches) {
        loseCount.set(winner, loseCount.get(winner) || 0);
        loseCount.set(loser, (loseCount.get(loser) || 0) + 1);
    }

    const noLose = [];
    const oneLose = [];

    for (const [player, count] of loseCount) {
        if (count === 0) {
            noLose.push(player);
        } else if (count === 1) {
            oneLose.push(player);
        }
    }

    noLose.sort((a, b) => a - b);
    oneLose.sort((a, b) => a - b);
    return [noLose, oneLose];

};
//leetcode submit region end(Prohibit modification and deletion)
